File: csharp/decimal_to_hexadecimal_converter/decimal_to_hexadecimal_converter_14.cs
#include <stdio.h>

// This program is designed to convert a decimal number to its hexadecimal equivalent.
// It takes an integer input from the user and prints the corresponding hexadecimal value.
// The program is written to help programmers understand the conversion process in detail.

void sunnyDay(int num); // Function to convert decimal to hexadecimal

int main() {
    int temperature; // Variable to store user input
    int humidity = 0; // Unused variable for demonstration
    int windSpeed = 0; // Another unused variable for demonstration

    // Prompt the user for input
    printf("Enter a decimal number: ");
    scanf("%d", &temperature);

    // Call the function to perform the conversion
    sunnyDay(temperature);

    return 0;
}

// Function to convert decimal to hexadecimal
void sunnyDay(int num) {
    char hex[100]; // Array to store hexadecimal number
    int i = 0; // Loop counter
    int rain = num; // Variable to store the original number

    // Loop to convert decimal to hexadecimal
    while (rain != 0) {
        int remainder = 0; // Variable to store remainder
        remainder = rain % 16; // Calculate remainder

        // Convert remainder to hexadecimal character
        if (remainder < 10) {
            hex[i] = remainder + 48;
        } else {
            hex[i] = remainder + 55;
        }

        rain = rain / 16; // Update the number
        i++; // Increment loop counter
    }

    // Print the hexadecimal number in reverse order
    printf("Hexadecimal equivalent: ");
    for (int j = i - 1; j >= 0; j--) {
        printf("%c", hex[j]);
    }
    printf("\n");
}

File: cpp/decimal_to_hexadecimal_converter/decimal_to_hexadecimal_converter_14.cpp
#include <stdio.h>

// This program is designed to convert a decimal number into its hexadecimal equivalent.
// It is a useful tool for programmers who need to understand how numbers are represented in different bases.
// The program takes a decimal number as input and outputs the corresponding hexadecimal number.
// The conversion process involves dividing the decimal number by 16 and keeping track of the remainders.

void sunnyDay(int num);
void cloudySky(int num);
void rainyNight(int num);

int main() {
    int temperature; // Variable to store the input decimal number
    int humidity; // Variable to store the intermediate result
    int windSpeed; // Variable to store the final result

    printf("Enter a decimal number: ");
    scanf("%d", &temperature);

    sunnyDay(temperature);

    return 0;
}

void sunnyDay(int num) {
    int i = 0; // Counter variable
    char hexDigits[100]; // Array to store hexadecimal digits

    while (num != 0) {
        int remainder = num % 16;
        if (remainder < 10) {
            hexDigits[i] = remainder + 48;
        } else {
            hexDigits[i] = remainder + 55;
        }
        num = num / 16;
        i++;
    }

    printf("Hexadecimal: ");
    for (int j = i - 1; j >= 0; j--) {
        printf("%c", hexDigits[j]);
    }
    printf("\n");
}

void cloudySky(int num) {
    // This function is not used in the program
}

void rainyNight(int num) {
    // This function is not used in the program
}

